packages/ui-extensions/src/surfaces/admin/components/CustomerSegmentTemplate/CustomerSegmentTemplate.ts

```diff
@@ -0,0 +1,33 @@
+import {createRemoteComponent} from '@remote-ui/core';
+
+/**
+ * Reserved namespace and key for the customer standard metafield used in the template's query.
+ * More info - https://shopify.dev/docs/apps/custom-data/metafields/definitions/standard
+ */
+type CustomerStandardMetafieldDependency = 'facts.birth_date';
+
+export interface CustomerSegmentTemplateProps {
+  /* Localized title of the template. */
+  title: string;
+  /* Localized description(s) of the template. */
+  description: string | string[];
+  /* ShopifyQL code snippet to render in the template with syntax highlighting. */
+  templateQuery: string;
+  /* ShopifyQL code snippet to insert in the segment editor. If missing, `templateQuery` will be used. */
+  templateQueryToInsert?: string;
+  /* List of customer standard metafields or custom metafields used in the template's query. */
+  dependencies?: {
+    standardMetafields: CustomerStandardMetafieldDependency[];
+    customMetafields: string[];
```

Comment:
We should have them both as optional if not adding one of the two requires the other to be there as well:

```suggestion
    standardMetafields?: CustomerStandardMetafieldDependency[];
    customMetafields?: string[];
```

Now a nit: doing the above would make `dependencies: {}` valid which if ok but  we could go crazy about it.

See [TS playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgApwK4BtkG8CwAUMicgA6ZYD8AXMgM5hSgDmA2gLoDcRAvkUVCRYiFADEocEAGs8RUshiSZtBk1acehfoUHho8JMgDCWTABMUBYqRBwAthDqNmIFloVgA9vdUAlCABHDGAoCHMAHnRsZAAfZAkpaQA+PgFdQgQvEEY1CAA3CBA6UwsUAF45GxI7RzoAcgBlSEKQeoAaeVJvezpcHV4gA)

```ts
interface Paul {
    paul?: string[];
}

interface Frank {
    frank?: string[];
}

interface Claude {
    name: string;
    tom?: Required<Paul | Frank>
}
```

Reply:
good catch! I made them both optional within it 👍